File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.ArrayDeque;
import java.util.Arrays;

public class Solution {
	public static String solution(String[] cards1, String[] cards2, String[] goal) {
		ArrayDeque<String> queGoal = new ArrayDeque<String>(Arrays.asList(goal));
		ArrayDeque<String> queC1 = new ArrayDeque<String>(Arrays.asList(cards1));
		ArrayDeque<String> queC2 = new ArrayDeque<String>(Arrays.asList(cards2));

		while (!queGoal.isEmpty()) {
			if ((!queC1.isEmpty() && queC1.peekFirst().equals(queGoal.peekFirst()))) {
				queC1.pollFirst();
				queGoal.pollFirst();
			} else if (!queC2.isEmpty() && queC2.peekFirst().equals(queGoal.peekFirst())) {
				queC2.pollFirst();
				queGoal.pollFirst();
			} else
				return "No";
		}

		if (queGoal.isEmpty())
			return "Yes";

		return "No";
	}
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <deque>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    deque <string> deque1(cards1.begin(), cards1.end());
    deque <string> deque2(cards2.begin(), cards2.end());

    string answer = "Yes";
    
    for (string str : goal)
    {
        if (deque1.front() == str) {
            deque1.pop_front();
        }
        else if (deque2.front() == str) {
            deque2.pop_front();
        }
        else {
            return "No";
        }
    }

    return answer;
}